src/server/gdb_server.c

```diff
@@ -2966,6 +2971,11 @@ static int gdb_query_packet(struct connection *connection,
 		gdb_connection->noack_mode = 1;
 		gdb_put_packet(connection, "OK", 2);
 		return ERROR_OK;
+	} else if (target_supports_gdb_query_custom(target)) {
```

Comment:
Why not just  check, if the function ptr is not null?  null fptr check is a general usage in OpenOCD.

Reply:
I was avoiding dereferencing target->type in a non-target module, but it looks like this is done at gdb_server.c:2994, so I will change it.  Thanks.